core/src/test/java/be/orbinson/aem/dictionarytranslator/servlets/action/ImportDictionaryServletTest.java

```diff
@@ -0,0 +1,106 @@
+
+        // Set request parameters
+        request.addRequestParameter("dictionary", "/content");
+        String csvData = "Labelname,EN,NL\nhello,Hello,Hallo\n";
+        InputStream csvStream = new ByteArrayInputStream(csvData.getBytes());
+        request.addRequestParameter("./csvfile", csvStream.readAllBytes(), "text/csv", "translations.csv");
+
+        // Call servlet's doPost method
+        importTranslation.doPost(request, response);
+
+        // Assert successful response
+        assertEquals(200, response.getStatus());
+
+        // Get the resourceResolver and assert that the nodes have been created
+        ResourceResolver resourceResolver = context.resourceResolver();
+        Resource enHelloResource = resourceResolver.getResource("/content/EN/hello");
+        assertNotNull(enHelloResource);
+
+        // Check the value of the sling:message property
+        ValueMap properties = enHelloResource.getValueMap();
+        String message = properties.get("sling:message", String.class);
+        assertEquals("Hello", message);
+
+        Resource nlHelloResource = resourceResolver.getResource("/content/NL/hello");
+        assertNotNull(nlHelloResource);
+
+        // Check the value of the sling:message property for Dutch
+        properties = nlHelloResource.getValueMap();
+        message = properties.get("sling:message", String.class);
+        assertEquals("Hallo", message);
+    }
+
+    @Test
+    void testDoPost_invalidCSV_WrongHeader() throws Exception {
+        MockSlingHttpServletRequest request = context.request();
+        MockSlingHttpServletResponse response = context.response();
+
+        // Set request parameters
+        request.addRequestParameter("dictionary", "/content");
+        String csvData = "WRONGHEADER,EN,NL\nhello,Hello,Hallo\n";
+        InputStream csvStream = new ByteArrayInputStream(csvData.getBytes());
+        request.addRequestParameter("./csvfile", csvStream.readAllBytes(), "text/csv", "translations.csv");
+
+        // Call servlet's doPost method
+        importTranslation.doPost(request, response);
+
+        // Assert successful response
+        assertEquals(400, response.getStatus());
+    }
+
+    @Test
+    void testDoPost_invalidCSV_UnsupportedDelimiter() throws Exception {
+        MockSlingHttpServletRequest request = context.request();
+        MockSlingHttpServletResponse response = context.response();
+
+        // Set request parameters
+        request.addRequestParameter("dictionary", "/content");
+        String csvData = "Labelname-EN-NL\nhello-Hello-Hallo\n";
+        InputStream csvStream = new ByteArrayInputStream(csvData.getBytes());
+        request.addRequestParameter("./csvfile", csvStream.readAllBytes(), "text/csv", "translations.csv");
+
+        // Call servlet's doPost method
+        importTranslation.doPost(request, response);
+
+        // Assert successful response
+        assertEquals(400, response.getStatus());
+    }
+}
```

Comment:
test with only header?

Reply:
added test with only a header, it is a valid csv according to the feature request, obviously no labels are added. Nothing changes